src/main/java/frc/robot/subsystems/drive/Drivetrain.java

```diff
@@ -207,21 +256,56 @@ private void altDrive(Vector spdVec, Vector rotVec) {
     move(spdVec, rot);
   }
 
+  /**
+   * returns the current speed of the robot from it's reference frame
+   *
+   * @return the current speed of the robot from it's reference frame
+   */
+  public ChassisSpeeds getRobotRelativeSpeeds() {
+    return DriveConstants.kDriveKinematics.toChassisSpeeds(
+        new SwerveModuleState[] {
```

Comment:
Wasn't there a simplified version you wrote earlier? Are they unrelated?

Reply:
I don't think so? I might have done a chassis speeds update based on odometry at one point, but it would be less accurate than kinematics from the swerve module states